### ISSUE TITLE
Type annotation ignored in `StarDist2D`

### ISSUE THREAD

**Issue by bersbersbers** [PyCQA/pylint#4788]
### Bug description

```python
"""Use pip install stardist."""
from stardist.models import StarDist2D

model: StarDist2D = StarDist2D.from_pretrained("2D_versatile_he")
model.predict_instances()
```

### Command used

```shell
pylint bug.py
```

### Pylint output

```shell
************* Module bug
bug.py:5:0: E1101: Instance of 'BaseModel' has no 'predict_instances' member (no-member)

--------------------------------------------------------------------
Your code has been rated at -6.67/10 (previous run: -6.67/10, +0.00)
```

### Expected behavior

No error.

### Pylint version

```shell
pylint 2.9.6
astroid 2.6.5
Python 3.9.6 (tags/v3.9.6:db3ff76, Jun 28 2021, 15:26:21) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

Windows 10

### Additional dependencies

stardist==0.7.2

**Comment by jacobtylerwalls** [comment]
`pylint` doesn't use type annotations to determine types (it's not a full-fledged type checker). This means you will run into false positives with `no-member` that we cannot address case by case. (`from_pretrained` doesn't seem to be in `stardist` anymore, either.)
